classes/helper/HelperList.php

```diff
@@ -483,11 +483,13 @@ public function displayEditLink($token, $id, $name = null)
             case 'index.php?controller=AdminProducts':
             case 'index.php?tab=AdminProducts':
                 // New architecture modification: temporary behavior to switch between old and new controllers.
-                global $kernel; // sf kernel
-                $pagePreference = $kernel->getContainer()->get('prestashop.core.admin.page_preference_interface');
-                $redirectLegacy = $pagePreference->getTemporaryShouldUseLegacyPage('product');
-                if (!$redirectLegacy && $this->identifier == 'id_product') {
-                    $href = Context::getContext()->link->getAdminLink('AdminProducts', true, ['id_product' => $id, 'updateproduct' => 1]);
+                $sfContainer = SymfonyContainer::getSfContainer();
+                if ($sfContainer) {
```

Comment:
You should not have to check, because we are in the admin environment, so `$kernel` must be defined